aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -1252,9 +1255,16 @@ Tensor& logsumexp_out(const Tensor& self, IntArrayRef dims, bool keepdim, Tensor
 }
 
 Tensor logsumexp(const Tensor& self, IntArrayRef dims, bool keepdim) {
-  Tensor result = at::empty({0}, self.options());
-  return at::native::logsumexp_out(self, dims, keepdim, result);
+  Tensor result;
+  if (at::isIntegralType(self.scalar_type(), /*includeBool=*/true)) {
```

Comment:
Hey @Kiyosora! Thanks for this PR. Sorry I'm just taking a look at it for the first time.

Would you update this promotion to cast self to the default floating point type and not always to float32? That is, if a user uses 

https://pytorch.org/docs/master/generated/torch.set_default_dtype.html?highlight=default%20dtype#torch.set_default_dtype

and sets double then this promotion should be to double. Taking a look at https://github.com/pytorch/pytorch/pull/34860/files may give you an idea for how to do this.